Library/ENV/4.3/cc

```diff
@@ -194,6 +196,27 @@ class Cmd
   end
 
   def keep?(path)
+    # The logic in this method will eventually become the default,
+    # but is currently opt-in.
+    return keep_orig?(path) unless ENV["HOMEBREW_EXPERIMENTAL_FILTER_FLAGS_ON_DEPS"]
+
+    # first two paths: reject references to Cellar or opt paths
+    # for unspecified dependencies
+    if path.start_with?(cellar)
+      %r[#{cellar}/([\w\-_\+]+)] =~ path
```

Comment:
Controversial idea here too: maybe we want to go all-in on mandating looking only in `opt` and not the Cellar? That's maybe a bad idea.